cameo/strain_design/deterministic/linear_programming.py

```diff
@@ -426,3 +430,394 @@ def _repr_html_(self):
+                model.solver = "gurobi"
+            # The tolerances are set to the minimum value. This gives maximum precision.
+            problem = model.solver.problem
+            problem.params.NodeMethod = 1  # primal simplex node relaxation
+            problem.params.FeasibilityTol = 1e-9
+            problem.params.OptimalityTol = 1e-3
+            problem.params.IntFeasTol = 1e-9
+            problem.params.MIPgapAbs = 1e-9
+            problem.params.MIPgap = 1e-9
+            # problem.params.TimeLimit = 200 # Use max 200 seconds when called, return best solution after that
+            # problem.params.PoolSearchMode = 1 #0 for only finding the optimum,
+            #      1 for finding more solutions (but no quality guaranteed), 2 for finding the n best possible solutions
+            # problem.params.PoolSolutions = 10 # Number of solutions kept when finding the optimal solution
+            # problem.params.PoolGap = 0.9 # only store solutions within 90% of the optimal objective value
+
+        elif "cplex" in cobra.util.solver.solvers.keys():
+            logger.warning(
+                "Changing solver to CPLEX, as Gurobi is not available."
+                "This may cause a big slowdown and limit options afterwards.")
+            if "cplex_interface" not in model.solver.interface.__name__:
+                model.solver = "cplex"
+            # The tolerances are set to the minimum value. This gives maximum precision.
+            problem = model.solver.problem
+            problem.parameters.mip.strategy.startalgorithm.set(1)  # primal simplex node relaxation
+            problem.parameters.simplex.tolerances.feasibility.set(
+                1e-9)
+            problem.parameters.simplex.tolerances.optimality.set(
+                1e-3)
+            problem.parameters.mip.tolerances.integrality.set(
+                1e-9)
+            problem.parameters.mip.tolerances.absmipgap.set(1e-9)
+            problem.parameters.mip.tolerances.mipgap.set(1e-9)
+            # problem.parameters.mip.pool.relgap.set(0.9)
+            #      For populate: find all solutions within 10% of the optimum for relgap = 0.1
+            # problem.parameters.timelimit.set(200) # Use max 200 seconds for solving
+            # problem.parameters.mip.limits.populate.set(20) # Find max 20 solutions (=default)
+
+        else:
+            logger.warning("You are trying to run 'GrowthCouplingPotential' with %s. This might not end well." %
+                           model.solver.interface.__name__.split(".")[-1])
+
+        # Remove reactions that are blocked: no flux through these reactions possible.
+        # This will reduce the search space for the solver, if not done already.
+        if remove_blocked:
+            blocked_reactions = cameo.flux_analysis.analysis.find_blocked_reactions(model)
+            model.remove_reactions(blocked_reactions)
+            blocked_reaction_ids = [r.id for r in blocked_reactions]
```

Comment:
Turning this into a set will improve the following statements.

```suggestion
            blocked_reaction_ids = {r.id for r in blocked_reactions}
```